File: Mechanica Engine/src/renderers/texture/TextureShader.java
package renderers.texture;

import renderers.WorldShader;

/**
 * Created by domin on 15 Apr 2017.
 */
public class TextureShader extends WorldShader {

    private static final String VERTEX_FILE = "texture/textureVertex.txt";
    private static final String FRAGMENT_FILE = "texture/textureFragment.txt";


    public TextureShader() {
        super(VERTEX_FILE, FRAGMENT_FILE);
    }

    @Override
    protected void getAllUniformLocations() {
        getMatrixLocations();
    }

    @Override
    protected void bindAttributes() {
        super.bindAttribute(0, "position");
        super.bindAttribute(1, "textureCoords");

    }
}

File: lwjgl3-test/src/shaders/texture/TextureShader.kt
package shaders.texture

import display.Game
import renderer.WorldShader


/**
 * Created by domin on 15 Apr 2017.
 */
class TextureShader : WorldShader("texture/textureVertex.glsl", "texture/textureFragment.glsl") {

    init {
        start()
        loadProjectionMatrix(Game.projectionMatrix.create())
        loadViewMatrix(Game.viewMatrix.create())
        stop()
    }

    override fun getAllUniformLocations() {
        getMatrixLocations()
    }

    override fun bindAttributes() {
        super.bindAttribute(0, "position")
        super.bindAttribute(1, "textureCoords")

    }



}
